src/service/cache.ts

```diff
@@ -6,7 +6,7 @@ const DAY = 60 * 60 * 24
 const CACHE = new Cache({ stdTTL: DAY })
 
 export interface Cachable<T extends string = any> {
-  value?: number | Array<T> | Record<T, string | number> | null
+  value?: number | Array<T> | Record<T, string | number> | null | Map<string, number>
```

Comment:
Probably will need to do `Map<string, BigNumber>`, based on my previous suggestion of keeping the holdings as BigNumbers.